### PR TITLE
Fix/overlays focus

### PR DESCRIPTION
**Pull request by jorenbroekema**
## What I did

1. Fix an issue with events no longer working inside the contentNode of local overlays due to body inert
2. Fix an issue with deepContains not properly handling multiple nested shadow roots, and fix a recursion problem where earlier loops would find a target, but later loops would not, and then containsTarget bool would end up as "false" even though we did find a target

### CODE REVIEW

File: packages/overlays/src/utils/deep-contains.js [hunk 1]
```diff
@@ -34,7 +32,7 @@ export function deepContains(el, targetEl) {
         return true;
       }
     }
-    checkChildren(el);
   }
+  checkChildren(el);
```
Comment (tlouisse) [inline packages/overlays/src/utils/deep-contains.js:36]: Difference is very subtle it seems. So basically we stopped traversal for `Element`s that are not `HTMLElement `?

Maybe we can combine the ifs on line 30 and 31?

File: packages/overlays/test/utils-tests/deep-contains.test.js [hunk 2]
```diff
@@ -63,4 +63,42 @@ describe('deepContains()', () => {
...
+    expect(deepContains(element, element.firstElementChild.shadowRoot)).to.be.true;
+    expect(deepContains(element, element.firstElementChild.shadowRoot.children[1])).to.be.true;
+    expect(deepContains(element, element.firstElementChild.shadowRoot.children[1].shadowRoot)).to.be
+      .true;
+    expect(
+      deepContains(
+        element,
+        element.firstElementChild.shadowRoot.children[1].shadowRoot.lastElementChild,
+      ),
+    ).to.be.true;
```
Comment (tlouisse) [inline packages/overlays/test/utils-tests/deep-contains.test.js:102]: Good test. Maybe also add a `to.be.false` to make sure we didn't implement `deepContains` as `() => true` ? :)

**Review by tlouisse — APPROVED** [review]

File: packages/overlays/src/utils/deep-contains.js (hunk 1 above)
Comment (jorenbroekema) [reply packages/overlays/src/utils/deep-contains.js:34]: Yes it stopped traversal for the ones that were ShadowRoot, and of course those should be traversed as well :D. If's can now be combined indeed, changed it 👍

File: packages/overlays/test/utils-tests/deep-contains.test.js (hunk 2 above)
Comment (jorenbroekema) [reply packages/overlays/test/utils-tests/deep-contains.test.js:108]: I added two tests to assert that deepContains returns false. One for siblings, one for when two nodes are just not related in any way (e.g. a sibling and its siblings child node, I guess a cousin node xD?)

File: packages/overlays/src/utils/deep-contains.js (hunk 1 above)
Comment (tlouisse) [reply packages/overlays/src/utils/deep-contains.js:34]: Yeah, I just figured it out indeed :) Ok, nice

File: packages/overlays/test/utils-tests/deep-contains.test.js (hunk 2 above)
Comment (tlouisse) [reply packages/overlays/test/utils-tests/deep-contains.test.js:108]: Haha, a cousin node I would say indeed. Those tests should be enough, so ready to merge I would say.

**Pull request merged**